### PR TITLE
Add IPv6 address when calculating target endpoints within L7EndpointsCalculator

### PR DESCRIPTION
**Pull request by gauravkghildiyal**
/assign @swetharepakula

### CODE REVIEW

File: pkg/neg/syncers/utils.go [hunk 1]
```diff
@@ -269,23 +276,25 @@ func toZoneNetworkEndpointMap(eds []negtypes.EndpointsData, zoneGetter negtypes.
...
-				// increment the count for duplicate endpoint
-				if existingPod, contains := networkEndpointPodMap[networkEndpoint]; contains {
-					dupCount += 1
-					if existingPod.Name < endpointAddress.TargetRef.Name {
-						continue // if existing name is alphabetically lower than current one, continue and don't replace
-					}
+			podIPs := ipsForPod[types.NamespacedName{Namespace: endpointAddress.TargetRef.Namespace, Name: endpointAddress.TargetRef.Name}]
+			networkEndpoint := negtypes.NetworkEndpoint{IP: podIPs.IP, Port: matchPort, Node: *endpointAddress.NodeName}
+			if enableDualStackNEG {
+				networkEndpoint.IPv6 = parseIPAddress(podIPs.IPv6)
```
Comment (swetharepakula) [inline pkg/neg/syncers/utils.go:282]: nit, add a comment here as well as to why we are doing this to make code readability better.

File: pkg/neg/syncers/utils_test.go [hunk 2]
```diff
@@ -1526,7 +1663,47 @@ func TestValidatePod(t *testing.T) {
...
 		t.Run(tc.desc, func(t *testing.T) {
 			if got := validatePod(tc.pod, nodeLister); got != tc.expect {
-				t.Errorf("validatePod() = %t, expected %t", got, tc.expect)
+				t.Errorf("validatePod() = %t, expected %t\n", got, tc.expect)
+			}
+		})
+	}
+}
+
+func TestParseIPv6Address(t *testing.T) {
```
Comment (swetharepakula) [inline pkg/neg/syncers/utils_test.go:1672]: nit: TestParseIPAddress

**Review by swetharepakula — COMMENTED** [review]
A few nits, otherwise this LGTM.

/hold Because #2044 is under review and almost ready to merge as well, let's hold off on merging this until that one is in.

File: pkg/neg/syncers/utils.go (hunk 1 above)
Comment (gauravkghildiyal) [reply pkg/neg/syncers/utils.go:284]: Done!

File: pkg/neg/syncers/utils_test.go [hunk 3]
```diff
@@ -441,107 +442,116 @@ func TestEnsureNetworkEndpointGroup(t *testing.T) {
...
-		portName            string
-		endpointSets        map[string]negtypes.NetworkEndpointSet
-		expectMap           negtypes.EndpointPodMap
-		networkEndpointType negtypes.NetworkEndpointType
+		desc                       string
+		portName                   string
+		wantZoneNetworkEndpointMap map[string]negtypes.NetworkEndpointSet
+		wantNetworkEndpointPodMap  negtypes.EndpointPodMap
+		networkEndpointType        negtypes.NetworkEndpointType
+		enableDualStackNEG         bool
```
[the remark this answers is not in the document]
Comment (gauravkghildiyal) [reply pkg/neg/syncers/utils_test.go:454]: Done.

File: pkg/neg/syncers/utils_test.go (hunk 2 above)
Comment (gauravkghildiyal) [reply pkg/neg/syncers/utils_test.go]: Done!

**Comment by gauravkghildiyal** [comment]
> A few nits, otherwise this LGTM.
> 
> /hold Because #2044 is under review and almost ready to merge as well, let's hold off on merging this until that one is in.

Sure!

**Comment by gauravkghildiyal** [comment]
/unhold
Looks good now

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**